File: NOVEMBER-2025/08-11-2025/1611.rb
def minimum_one_bit_operations(n)
  ind = 0
  ans = []
  while n != 0
    if (n & 1) == 1
      temp = 2**ind
      temp = temp * 2
      temp -= 1
      ans.push(temp)
    end
    ind += 1
    n = n >> 1
  end
  res = 0
  ans.reverse!
  flag = true
  ans.each do |it|
    if flag
      res += it
      flag = false
    else
      res -= it
      flag = true
    end
  end
  res
end

File: NOVEMBER-2025/08-11-2025/1611.dart
class Solution {
  int minimumOneBitOperations(int n) {
      int multiplier = 1;
        int res = 0;
        while(n > 0) {
            res += n ^ (n-1) * multiplier;
            multiplier = -1 * multiplier;
            n &= n-1;
        }
        if(res>=0){
            return res;
        }else{
            return -1*res;
        }
  }
}